src/cli/tools/protoc.ts

```diff
@@ -0,0 +1,66 @@
+import path from 'node:path';
+import { execa } from 'execa';
+import { inject, injectable } from 'inversify';
+import { BaseInstallService } from '../install-tool/base-install.service';
+import {
+  CompressionService,
+  EnvService,
+  HttpService,
+  PathService,
+} from '../services';
+import { semverCoerce } from '../utils';
+
+@injectable()
+export class ProtocInstallService extends BaseInstallService {
+  readonly name = 'protoc';
+
+  private get ghArch(): string {
+    switch (this.envSvc.arch) {
+      case 'arm64':
+        return 'aarch_64';
+      case 'amd64':
+        return 'x86_64';
+    }
+  }
+
+  constructor(
+    @inject(EnvService) envSvc: EnvService,
+    @inject(PathService) pathSvc: PathService,
+    @inject(HttpService) private http: HttpService,
+    @inject(CompressionService) private compress: CompressionService,
+  ) {
+    super(pathSvc, envSvc);
+  }
+
+  override async install(version: string): Promise<void> {
+    const name = this.name;
+
+    const url = `https://github.com/protocolbuffers/protobuf/releases/download/v${version}/${name}-${version}-linux-${this.ghArch}.zip`;
+
+    const file = await this.http.download({
+      url,
+    });
+
+    const cwd = await this.pathSvc.createVersionedToolPath(name, version);
+
+    await this.compress.extract({ file, cwd });
+  }
+
+  override async link(version: string): Promise<void> {
+    const src = path.join(
+      await this.pathSvc.createVersionedToolPath(this.name, version),
```

Comment:
```suggestion
      await this.pathSvc.versionedToolPath(this.name, version),
```
path should exist here